### PR TITLE
feat: coze realtime api SDK

### PR DESCRIPTION
**Pull request by jackshen310**
## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive real-time communication SDK for voice interactions with Coze AI bots.
	- Added functionality for managing audio and connection states in a real-time console application.
	- Implemented a structured event handling system for real-time API operations.
	- Added new components for user interaction, including `MessageForm`, `ConsoleFooter`, and `Settings`.
	- Introduced a new `RealtimeClient` class for managing real-time audio interactions.
	- Added a new `EngineClient` class for handling audio streams and user interactions.
	- Enhanced environment variable management for dynamic application configuration.
	- Added a new MIT License for the `realtime-api` package.

- **Documentation**
	- Updated README.md to include installation instructions, core operations, and event handling guidance.
	- Added a new HTML entry point for the real-time console application.

- **Bug Fixes**
	- Enhanced error handling for connection issues and microphone permissions.

- **Tests**
	- Added extensive unit tests for core functionalities, including event management and real-time client operations.
	- Introduced tests for new components and utility functions.

### CODE REVIEW

File: packages/realtime-api/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+3. Low Latency: Implemented using RTC (Real-Time Communication) technology to minimize latency throughout the communication pipeline.
+
+4. Natural Voice Synthesis: Using advanced TTS (Text-to-Speech) models powered by large language models, our system:
+   - Intelligently predicts emotional context and intonation
+   - Generates ultra-natural, high-fidelity, personalized voice output
+   - Excels in naturalness, audio quality, prosody, breathing patterns, and emotional expression
+   - Handles mixed Chinese-English content seamlessly
+   - Delivers human-like expression of modal particles and emotional nuances
+
+![api-overview](./assets/api-overview.png)
```
[the remark this answers is not in the document]
Comment (jackshen3102) [reply packages/realtime-api/README.md:23]: fixed

**Review by jackshen3102 — APPROVED** [review]

File: packages/realtime-api/examples/realtime-console/src/ConsoleFooter.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+  const handleToggleMicrophone = () => {
+    onToggleMicrophone(!isMicrophoneOn);
+    if (clientRef.current) {
+      clientRef.current.setAudioEnable(!isMicrophoneOn);
+      message.success(`Microphone ${!isMicrophoneOn ? 'unmuted' : 'muted'}`);
+    } else {
+      message.error('Please set accessToken and botId first');
+    }
+  };
```
[the remark this answers is not in the document]
Comment (jackshen310) [reply packages/realtime-api/examples/realtime-console/src/ConsoleFooter.tsx]: fixed

File: packages/realtime-api/examples/realtime-console/src/ConsoleFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    } catch (e) {
+      if (e instanceof RealtimeAPIError) {
+        message.error(`Failed to interrupt: ${e.message}`);
+      } else {
+        message.error('Failed to interrupt');
+      }
+      console.error(e);
+      return;
+    }
+  };
```
[the remark this answers is not in the document]
Comment (jackshen310) [reply packages/realtime-api/examples/realtime-console/src/ConsoleFooter.tsx]: done

File: packages/realtime-api/src/client.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    console.log('handleRemoteAudioPropertiesReport', event);
+  }
+
+  async startAudioPlaybackDeviceTest() {
+    await this.engine.startAudioPlaybackDeviceTest('audio-test.wav', 200);
+  }
+
+  stopAudioPlaybackDeviceTest() {
+    this.engine.stopAudioPlaybackDeviceTest();
+  }
```
[the remark this answers is not in the document]
Comment (jackshen310) [reply packages/realtime-api/src/client.ts]: done

File: packages/realtime-api/src/client.ts [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+      this.engine.unpublishStream(MediaType.AUDIO);
+    }
+  }
+
+  async disconnect() {
+    await this.engine.stopAudioCapture();
+    await this.engine.unpublishStream(MediaType.AUDIO);
+    await this.engine.leaveRoom();
+    this.removeEventListener();
+  }
```
[the remark this answers is not in the document]
Comment (jackshen310) [reply packages/realtime-api/src/client.ts]: done

**Review by jsongo — APPROVED** [review]

**Pull request merged**